tests/periph/freqm/main.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2023 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Application to test functionality of the frequency meter
+ *              peripheral
+ *
+ * @author      Urs Gompper <urs.gompper@ml-pa.com>
+ * @}
+ */
+
+#include <stddef.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdint.h>
+
+#include "periph/freqm.h"
+
+#define UNUSED(x) (void)(x)
+
+volatile bool freqm_buffer_ready = false;
+extern struct freqm_config config_freqm;
+
+void freqm_event_handler(uintptr_t context)
+{
+    UNUSED(context);
+    if ((freqm_error_get() == 0) && (freqm_is_busy() == false)) {
+        freqm_buffer_ready = true;
+    }
+}
+
+int main( void )
+{
+    uint32_t measuredclock_freq;
+
+    puts("FREQM peripheral driver test");
+
+    /* Initialize frequency meter peripheral */
+    if (freqm_init(&config_freqm) != 0) {
+        return EXIT_FAILURE;
+    }
+
+    /* Register callback handler for FREQM interrupt */
+    freqm_callback_register(freqm_event_handler, (uintptr_t)NULL);
+
+    /* Start frequency measurement */
+    if (freqm_start() == false) {
+        return EXIT_FAILURE;
+    }
+
+    /* Wait for measurement to be finished*/
+    while (freqm_buffer_ready != true) {}
+
+    /* Read measuremant */
+    measuredclock_freq = freqm_frequency_get(&config_freqm);
+
+    printf("Measured Clock Frequency: %ld Hz\n\n", measuredclock_freq);
```

Comment:
Do we really need the double newline ?

Reply:
No, this was just my personal preference. I can also live with it like this:
```suggestion
    printf("Measured Clock Frequency: %ld Hz\n", measuredclock_freq);
```